apps/condo/domains/miniapp/schema/B2BAccessToken.js

```diff
@@ -0,0 +1,282 @@
+    }
+    if (!accessRightForUserAndApp) {
+        return ERRORS.SERVICE_USER_DOES_NOT_MATCH_CONTEXT
+    }
+    return null
+}
+
+const ERRORS = {
+    CONTEXT_NOT_EXISTS: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_CONTEXT_NOT_EXISTS,
+        message: 'B2BAppContext does not exists',
+    },
+    RIGHT_SET_NOT_EXISTS: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_RIGHT_SET_NOT_EXISTS,
+        message: 'B2BAppAccessRightSet does not exists',
+    },
+    WRONG_RIGHT_SET_TYPE: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_WRONG_RIGHT_SET_TYPE,
+        message: 'Wrong B2BAppAccessRightSet type, must be "SCOPED"',
+    },
+    CONTEXT_DOES_NOT_MATCH_RIGHT_SET: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_CONTEXT_DOES_NOT_MATCH_RIGHT_SET,
+        message: 'B2BAppContext and B2BRightSet connected to different B2BApps',
+    },
+    SERVICE_USER_DOES_NOT_MATCH_CONTEXT: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_SERVICE_USER_DOES_NOT_MATCH_CONTEXT,
+        message: 'User and B2BAppContext connected to different B2BApps',
+    },
+    NEGATIVE_TTL: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_NEGATIVE_TTL,
+        message: 'TTL can\'t be negative. You provided expiresAt datetime previous to the current one',
+    },
+}
+
+const ERRORS_FOR_DELETION = new Set([
+    ERRORS.CONTEXT_NOT_EXISTS,
+    ERRORS.RIGHT_SET_NOT_EXISTS,
+    ERRORS.NEGATIVE_TTL,
+])
+
+const B2BAccessToken = new GQLListSchema('B2BAccessToken', {
+    schemaDoc: 'Access to service user with additional restrictions by Organization and B2BAppAccessRightSet',
+    fields: {
+        sessionId: {
+            schemaDoc: 'Encrypted sessionId of session',
+            type: 'EncryptedText',
+            isRequired: true,
+            access: access.updatableOnlyForAdmin,
+            hooks: {
+                resolveInput ({ operation, existingItem, fieldPath, originalInput }) {
+                    const inputSessionId = get(originalInput, fieldPath)
+                    // need to have option to manually set sessionId, so we can migrate already given tokens to this
+                    if (!isNil(inputSessionId) && !isEmpty(inputSessionId)) {
+                        return inputSessionId
+                    }
+                    if (operation === 'create') {
+                        return ACCESS_TOKEN_SESSION_ID_PREFIX + crypto.randomUUID()
+                    }
+                    return get(existingItem, fieldPath)
+                },
+            },
+        },
+
+        token: {
+            schemaDoc: 'Token, ready to be put in Authorization header. Shown only once after creation',
+            type: 'Virtual',
+            graphQLReturnType: 'String',
+            resolver: (item) => {
+                if (!isNil(item.token)) {
+                    return item.token
+                }
+                return null
+            },
```

Comment:
access?

Reply:
I found out that in case of virtual field, only 'read' access is called. And it does not contain information about operation (creating entity, 'create' access is not called, only 'read').

So access for virtual field does nothing, aside of User checks, which defined by list access